content/architecture/use-modules-to-define-features.md

```diff
@@ -0,0 +1,114 @@
+---
+title: use modules to define features
+author: 
+  name: Billy Lando
+  url: https://github.com/billyjov
+---
+
+# Problem 
+
+As an app scales, the root module start growing. Loading multiple components increase the start time and affect the application performance.    
+The more components added, the more confusing it will be to maintain the entire application.
+
+# Solution 
+
+Module in Angular is a mechanism to group and encapsulate components, services, directives and pipes.   
```

Comment:
Found some typos and grammatical errors. I propose the following:

```
Using an `NgModule` to define features in Angular allows lazy loading, isolation and portability of this features. The purpose of feature modules is to organize relevant code into cohesive blocks. This helps to apply clear boundaries between features or application domains.

Another benefit of `NgModules` is that they allow us to lazy load each feature module in order to boost the performance of our app, especially time to **first meaningful paint** (FMP). Furthermore, feature encapsulation using modules allow us to replace each module without (or with small changes) affecting other modules.
```

Generally for texts, we should aim for `we/us` instead of `I/you`. In my opinion, this makes the reader feel more welcome and it feels like we are doing / reading this together.

Reply:
sure you are right..i'll change this...thanks for your improvments and your time